### ISSUE TITLE
RAPS16 : Unlimited polymorphic type does not work with transfer intrinsic

### ISSUE THREAD

**Issue by kiranchandramohan** [flang-compiler/flang#226]
This issue affects the RAPS16 benchmark. A simple testcase to reproduce the issue is given below. Flang rejects this valid and conforming fortran program with the error given below. This program is rejected probably due to lack of support for unlimited polymorphic type in the transfer intrinsic.

F90-S-0151-Empty TYPE, STRUCTURE, UNION, or MAP (tests/simple.f90: 6)
  0 inform,   0 warnings,   1 severes, 0 fatal for simple

program simple
  integer, target :: tar = 10
  class (*), pointer :: ptr
  integer val
  ptr => tar
  val = transfer(ptr,1)
  print *, val
end program

PS : Interestingly, it seems Intel fortran compiler hit the same error last year.
https://software.intel.com/en-us/forums/intel-fortran-compiler-for-linux-and-mac-os-x/topic/622171

**Comment by pawosm-arm** [comment]
The code above can be compiled with flang patched with sick hack like this:
```
-    ARGT_ARG(newargt, 3) = size_ast_of(srcarray, DDTG(A_DTYPEG(srcarray)));
+    ARGT_ARG(newargt, 3) = (DT_RSVD1 < DDTG(A_DTYPEG(srcarray))) ? size_ast(sym_of_ast(mask), DDTG(A_DTYPEG(mask))) : size_ast_of(srcarray, DDTG(A_DTYPEG(srcarray)));
```

**Comment by gklimowicz** [comment]
This should be fixed in PR#452. @kiranchandramohan let me know if this is not the case.

**Comment by kiranchandramohan** [comment]
I can confirm that this works fine on current master. I did not specifically check whether it was fixed with PR#452.
Thanks for fixing the issue @gklimowicz.
